spring-amqp/src/main/java/org/springframework/amqp/support/converter/DefaultClassMapper.java

```diff
@@ -45,27 +49,61 @@
 
 	private volatile Map<Class<?>, String> classIdMapping = new HashMap<Class<?>, String>();
 
-	private volatile Class<?> defaultHashtableClass = Hashtable.class;
+	private volatile Class<?> defaultMapClass = LinkedHashMap.class;
 
 	private volatile Class<?> defaultType;
 
 	/**
 	 * The type returned by {@link #toClass(MessageProperties)} if no type information
 	 * is found in the message properties.
-	 * @param defaultType the defaultType to set
+	 * @param defaultType the defaultType to set.
 	 */
 	public void setDefaultType(Class<?> defaultType) {
 		this.defaultType = defaultType;
 	}
 
-	public void setDefaultHashtableClass(Class<?> defaultHashtableClass) {
-		this.defaultHashtableClass = defaultHashtableClass;
+	/**
+	 * Set the type of {@link Map} to use. For outbound messages, set the
+	 * {@value #DEFAULT_CLASSID_FIELD_NAME} header to {@code HashTable}. For inbound messages,
+	 * if the {@value #DEFAULT_CLASSID_FIELD_NAME} header is {@code HashTable} convert to thi
+	 * class.
+	 * @param defaultMapClass the map class.
+	 * @deprecated use {@link #setDefaultMapClass(Class)}
+	 */
+	@Deprecated
+	public void setDefaultHashtableClass(Class<?> defaultMapClass) {
+		this.defaultMapClass = defaultMapClass;
+	}
+
+	/**
+	 * Set the type of {@link Map} to use. For outbound messages, set the
+	 * {@value #DEFAULT_CLASSID_FIELD_NAME} header to {@code HashTable}. For inbound messages,
+	 * if the {@value #DEFAULT_CLASSID_FIELD_NAME} header is {@code HashTable} convert to this
```

Comment:
I also think we could set the `defaultType` to `LinkedHashMap` - to prevent the...

	throw new MessageConversionException(
			"failed to convert Message content. Could not resolve "
					+ getClassIdFieldName() + " in header " +
					"and no defaultType provided");

again, to be consistent with Jackson